geomstats/geometry/grassmannian.py

```diff
@@ -50,9 +50,33 @@ def belongs(self, point, tolerance=TOLERANCE):
         -------
         belongs : bool
         """
-        raise NotImplementedError(
-            'The Grassmann `belongs` is not implemented.'
-            'It shall test whether p*=p, p^2 = p and rank(p) = k.')
+        point = gs.to_ndarray(point, to_ndim=3)
+        n_points, n, p = point.shape
+
+        # check square
```

Comment:
We try to avoid comments in the codebase, because they are not sustainable: it's easy to change a piece of code and forget to update the comment, and soon nothing makes sense.

To organize this code, how about creating auxiliary private methods: _check_square, _check_idempotency, _check_rank?